src/main/java/uk/gov/ons/ssdc/caseprocessor/messaging/NewCaseReceiver.java

```diff
@@ -29,26 +29,27 @@
 
 @MessageEndpoint
 public class NewCaseReceiver {
+
   private final CaseRepository caseRepository;
   private final CaseService caseService;
   private final CollectionExerciseRepository collectionExerciseRepository;
   private final EventLogger eventLogger;
   private final RasRmCaseNotificationEnrichmentService rasRmNewBusinessCaseEnricher;
-
-  @Value("${caserefgeneratorkey}")
-  private byte[] caserefgeneratorkey;
+  private final byte[] caserefgeneratorkey;
 
   public NewCaseReceiver(
       CaseRepository caseRepository,
       CaseService caseService,
       CollectionExerciseRepository collectionExerciseRepository,
       EventLogger eventLogger,
-      RasRmCaseNotificationEnrichmentService rasRmNewBusinessCaseEnricher) {
+      RasRmCaseNotificationEnrichmentService rasRmNewBusinessCaseEnricher,
+      @Value("${caserefgeneratorkey}") byte[] caserefgeneratorkeybytes) {
     this.caseRepository = caseRepository;
     this.caseService = caseService;
     this.collectionExerciseRepository = collectionExerciseRepository;
     this.eventLogger = eventLogger;
     this.rasRmNewBusinessCaseEnricher = rasRmNewBusinessCaseEnricher;
+    this.caserefgeneratorkey = caserefgeneratorkeybytes.clone();
```

Comment:
Why these changes?  It seemed to work fine previously and this make the test set up more awkward.